From 9392963f02cc0d3c6cbe852edb2bc34f746afd2e Mon Sep 17 00:00:00 2001
From: Guanhua Gao <guanhua.gao@nxp.com>
Date: Mon, 26 Aug 2019 10:36:59 +0800
Subject: [PATCH 062/245] staging: fsl-dpaa2/ethsw: Remove some paramters to
 backward compatible with linux-4.14

commit c359f94af00ea30a3f626fc0d0c61f7fe0c42615 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux.git

Remove some parameters to backward compatible with linux-4.14.

Signed-off-by: Guanhua Gao <guanhua.gao@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/staging/fsl-dpaa2/ethsw/ethsw.c | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/ethsw/ethsw.c b/drivers/staging/fsl-dpaa2/ethsw/ethsw.c
index a9c3b60195c9..5bb5f96e74f0 100644
--- a/drivers/staging/fsl-dpaa2/ethsw/ethsw.c
+++ b/drivers/staging/fsl-dpaa2/ethsw/ethsw.c
@@ -318,8 +318,7 @@ static int ethsw_port_fdb_del_mc(struct ethsw_port_priv *port_priv,
 
 static int port_fdb_add(struct ndmsg *ndm, struct nlattr *tb[],
 			struct net_device *dev, const unsigned char *addr,
-			u16 vid, u16 flags,
-			struct netlink_ext_ack *extack)
+			u16 vid, u16 flagsi)
 {
 	if (is_unicast_ether_addr(addr))
 		return ethsw_port_fdb_add_uc(netdev_priv(dev),
@@ -1196,8 +1195,6 @@ static void ethsw_switchdev_event_work(struct work_struct *work)
 
 	switch (switchdev_work->event) {
 	case SWITCHDEV_FDB_ADD_TO_DEVICE:
-		if (!fdb_info->added_by_user)
-			break;
 		if (is_unicast_ether_addr(fdb_info->addr))
 			err = ethsw_port_fdb_add_uc(netdev_priv(dev),
 						    fdb_info->addr);
@@ -1206,13 +1203,10 @@ static void ethsw_switchdev_event_work(struct work_struct *work)
 						    fdb_info->addr);
 		if (err)
 			break;
-		fdb_info->offloaded = true;
 		call_switchdev_notifiers(SWITCHDEV_FDB_OFFLOADED, dev,
-					 &fdb_info->info, NULL);
+					 &fdb_info->info);
 		break;
 	case SWITCHDEV_FDB_DEL_TO_DEVICE:
-		if (!fdb_info->added_by_user)
-			break;
 		if (is_unicast_ether_addr(fdb_info->addr))
 			ethsw_port_fdb_del_uc(netdev_priv(dev), fdb_info->addr);
 		else
-- 
2.17.1

